### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by naxvog**
Browsed around some Flarum components and integrated some of them to increase general standardization of the extension:
- LoadingIndicator component usage to improve the look and feel
- Loading class for the component wrapper
- Icon component to improve accessibility
- Preview generation on `refreshContent` to include post updates as well (triggers right after a post was edited without a page refresh)
- General refactoring

### CODE REVIEW

**Comment by dat-archielite** [comment]
Could you delete the `dist` folder?

**Comment by datlechin** [comment]
Could you delete the dist folder?

**Comment by datlechin** [comment]
Thank you <3

**Review by datlechin — APPROVED** [review]

File: js/src/forum/components/LinkPreview.js [hunk 1]
```diff
@@ -42,7 +49,7 @@ export default class LinkPreview extends Component {
   }
 
   getLink(text) {
-    return m('a', this.linkAttributes, text);
+    return <Link {...this.linkAttributes}>{this.loading ? this.getDomain() : text ?? this.getDomain()}</Link>;
```
Comment (datlechin) [inline js/src/forum/components/LinkPreview.js:52]: There's a problem here that I didn't notice before, the `Link` component cannot be redirected to the destination page.

Comment (naxvog) [reply js/src/forum/components/LinkPreview.js:52]: Could you please elaborate on that using an example? I tested the Component and found no issue on my dev installation.

**Pull request merged**